### DISCUSSION TITLE
Renaming "batch" to "bulk"

### DISCUSSION THREAD

**Discussion by ldiebold** [tailflow/laravel-orion#272] [Ideas]
Hey there!

I've been working on a library that standardizes how a vue frontend works with an API. The first two drivers that will be supported are "indexedDB" and "Orion".

When working on the API, I discovered that the word "batch" tends to mean "many **different** kinds of operations in one request". In Orion, batch means "many operations **of the same kind** in one request". Does that make sense?

It's not a big deal, though may come back to bite this project if true "batch" operations are later added. For example, I can imagine Orion supporting an endpoint like this in the future:
```json
[
  {
    "operation": "batchCreate",
    "entity": "users",
    "resources": [
      {
        "name": "Luke"
      },
      {
        "email": "luke@blah.com"
      }
    ]
  },
  {
    "operation": "update",
    "entity": "users",
    "primaryKey": 1,
    "data": {
      "name": "johnno"
    }
  },
  {
    "operation": "batchUpdate",
    "entity": "posts",
    "resources": {
      "3": {
        "name": "Brian"
      }
    }
  }
]
```

Not the best example, but you get the idea!

So all I'm trying to point out, is that the example above is a "batch" request, and what Orion currently has implemented is a "bulk" request. Not a big deal and often devs use this language interchangeably, yet if "true batch" operations were later added, we'd have a naming problem.

By the way, I absolutely adore this package. For me, it's the by far the single most important Laravel package we use on every single project <3

**Comment by v-radev** [comment]
Hey, I'm interesting in learning more about batch with different kind of operations, can you share examples or docs for something like that? Until now I never thought batch could be different types of requests and I'm wondering how this might be used. Thanks

**Reply by ldiebold** [reply]
I haven't looked much into it myself, and would usually just send multiple requests. here's some [microsoft docs on batch requests using graph](https://learn.microsoft.com/en-us/graph/json-batching)

**Comment by alexzarbn** [comment]
I think it is a matter of preference rather than a standard 🤔

P.S. Claude's description of "batch" seems like a better fit in this case.
![Screenshot 2024-09-16 at 16 34 53](https://github.com/user-attachments/assets/7237e59a-2092-4e7a-8387-fdbb71ba3b22)

**Reply by ldiebold** [reply]
Oh wow, you're absolutely right.
Different sources have completely opposite opinions on what the two words mean!

I kinda wish it was a standard rather than an opinion. I'm going to do a bit more digging, then just choose whatever seems to be the most prevalent.

Thanks for getting back to me!
